drivers/gpio/gpio_litex.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2019 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <drivers/gpio.h>
+#include <zephyr/types.h>
+#include <sys/util.h>
+#include <string.h>
+#include <logging/log.h>
+
+#define SUPPORTED_FLAGS (GPIO_INPUT | GPIO_OUTPUT | \
+			GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH | \
+			GPIO_ACTIVE_LOW | GPIO_ACTIVE_HIGH)
+
+#define GPIO_LOW        0
+#define GPIO_HIGH       1
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+LOG_MODULE_REGISTER(gpio_litex);
+
+static const char *LITEX_LOG_REG_SIZE_NGPIOS_MISMATCH =
+	"Cannot handle all of the gpios with the register of given size\n";
+static const char *LITEX_LOG_WRONG_DIR =
+	"Direction chosen in device tree do not match with the operation\n";
+static const char *LITEX_LOG_CANNOT_CHANGE_DIR =
+	"Cannot change port direction selected in device tree\n";
+
+struct gpio_litex_cfg {
+	volatile u32_t *reg_addr;
+	int reg_size;
+	int nr_gpios;
+	bool port_is_output;
+};
+
+struct gpio_litex_data {
+	struct gpio_driver_data common;
+};
+
+/* Helper macros for GPIO */
+
+#define DEV_GPIO_CFG(dev)						\
+	((const struct gpio_litex_cfg *)(dev)->config->config_info)
+
+/* Helper functions for bit / port access */
+
+static inline void set_bit(const struct gpio_litex_cfg *config,
+			   u32_t bit, u32_t val)
+{
+	int regv, new_regv;
+
+	regv = litex_read(config->reg_addr, config->reg_size);
+	new_regv = (regv & ~BIT(bit)) | (val << bit);
```

Comment:
The path to this line through `gpio_pin_write()` appears to shift an unvalidated 32-bit value into the bits at and above the provided bit.  Since this is deprecated API I suppose that might not matter.  A simple solution would be to change the parameter type from `u32_t` to `bool`.

Reply:
Changed `val` argument type to `bool`.

Reply:
@pabigot Is it good now?

Reply:
Yes, thanks.